File: QadeerApp.Web/Modules/ServerTypes/Cv/EmployeeCvStatusColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeCvStatusRow } from "./EmployeeCvStatusRow";

export interface EmployeeCvStatusColumns {
    DisplayName: Column<EmployeeCvStatusRow>;
    DepartmentName: Column<EmployeeCvStatusRow>;
    SpecializationName: Column<EmployeeCvStatusRow>;
    IsCompleted: Column<EmployeeCvStatusRow>;
    CvUpdatedOn: Column<EmployeeCvStatusRow>;
    QualificationCount: Column<EmployeeCvStatusRow>;
    ExperienceCount: Column<EmployeeCvStatusRow>;
    CourseCount: Column<EmployeeCvStatusRow>;
}

export class EmployeeCvStatusColumns extends ColumnsBase<EmployeeCvStatusRow> {
    static readonly columnsKey = 'Cv.EmployeeCvStatus';
    static readonly Fields = fieldsProxy<EmployeeCvStatusColumns>();
}

File: QadeerApp.Web/Modules/Cv/EmployeeCvStatus/EmployeeCvStatusColumns.cs
namespace QadeerApp.Cv.Columns;

[ColumnsScript("Cv.EmployeeCvStatus")]
[BasedOnRow(typeof(EmployeeCvStatusRow), CheckNames = true)]
public class EmployeeCvStatusColumns
{
    [Width(200)]
    public string DisplayName { get; set; }

    [Width(140)]
    public string DepartmentName { get; set; }

    [Width(140)]
    public string SpecializationName { get; set; }

    [Width(110)]
    public bool IsCompleted { get; set; }

    [Width(130)]
    public DateTime CvUpdatedOn { get; set; }

    [Width(120)]
    public int QualificationCount { get; set; }

    [Width(120)]
    public int ExperienceCount { get; set; }

    [Width(120)]
    public int CourseCount { get; set; }
}
